### PR TITLE
[MM-44434] Introduced search for channel id's through system console.

### LINKED ISSUE #20252: Allow searching for channel IDs in the system console
Repro: Paste a channel ID in the search input field in the System Console > User Management > Channels page

Observed: no results

Expected: allow returning results based on the channel ID (currently it only matches on the channel name)

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-44434

### PR DESCRIPTION
**Pull request by santoniriccardo**
#### Summary

Introduced search for channel id's, not just channel name.

Screen Shot 2022-05-22 at 12 29 34 AM

#### Ticket Link

Fixes #20252

https://mattermost.atlassian.net/browse/MM-44434

#### Release Note

```release-note
- Added new parameter to channel search endpoint, 'IncludeSearchById', allowing requests to include searches that match IDs in response.
- Utilizing new channel search option to search db for matches on id.
- System console now searches and returns channels based on id as well.
```

#### Related Pull Requests
Included webapp changes, see https://github.com/mattermost/mattermost-webapp/pull/10359

### CODE REVIEW

**Review by neallred — APPROVED** [review]
LGTM! Thanks, @santoniriccardo . I validated by manually doing the request in dev tools with this parameter the mattermost-server instance (https://mattermost-server-pr-20259.test.mattermost.cloud/) since I don't *think* the build system will pick up both forks at the same time.

File: store/storetest/channel_store.go [hunk 1]
```diff
@@ -6436,6 +6436,7 @@ func testChannelStoreSearchAllChannels(t *testing.T, ss store.Store) {
 		{"Filter team 1 and team 2, public and private and group constrained", "", store.ChannelSearchOpts{IncludeDeleted: false, TeamIds: []string{t1.Id, t2.Id}, Public: true, Private: true, GroupConstrained: true, Page: model.NewInt(0), PerPage: model.NewInt(5)}, model.ChannelList{&o5}, 1},
 		{"Filter team 1 and team 2, public and private and exclude group constrained", "", store.ChannelSearchOpts{IncludeDeleted: false, TeamIds: []string{t1.Id, t2.Id}, Public: true, Private: true, ExcludeGroupConstrained: true, Page: model.NewInt(0), PerPage: model.NewInt(5)}, model.ChannelList{&o1, &o2, &o3, &o4, &o6}, 12},
 		{"Filter deleted returns only deleted channels", "", store.ChannelSearchOpts{Deleted: true, Page: model.NewInt(0), PerPage: model.NewInt(5)}, model.ChannelList{&o13}, 1},
+		{"Search ChannelA by id", o1.Id, store.ChannelSearchOpts{IncludeDeleted: false, IncludeSearchById: true}, model.ChannelList{&o1}, 0},
```
Comment (ashishbhate) [inline store/storetest/channel_store.go:6439]: Can we set Page to 0 and PerPage to something in the ChannelSearchOpts, to confirm that TotalCount is 1 in the test?

File: store/sqlstore/channel_store.go [hunk 2]
```diff
@@ -3279,14 +3279,27 @@ func (s SqlChannelStore) channelSearchQuery(opts *store.ChannelSearchOpts) sq.Se
...
+	}
+
+	likeClause, likeTerm := s.buildLIKEClause(opts.Term, likeFields)
+
+	// Keep the number of likeTerms same as the number of columns
+	// (c.Name, c.DisplayName, c.Purpose, c.Id?)
+	likeTerms := make([]interface{}, len(strings.Split(likeFields, ",")))
+	for i := 0; i < len(likeTerms); i++ {
+		likeTerms[i] = likeTerm
+	}
```
Comment (ashishbhate) [inline store/sqlstore/channel_store.go:3294]: This can be moved inside the conditional on line 3296.

**Review by ashishbhate — CHANGES_REQUESTED** [review]
Thanks @santoniriccardo! Looking good. Two small suggestions inline.

**Comment by santoniriccardo** [comment]
@ashishbhate Resolved!

**Review by ashishbhate — APPROVED** [review]
💯

**Review by jgilliam17 — COMMENTED** [review]
Edited: I missed the webapp PR that was linked. Will test there now

Thanks @santoniriccardo 
While testing on the PR test server, I wasn't able to search system console>user management>channels using channel id.
Screenshot 2022-06-14 154222
Can you please take a look? Thanks 🙂

**Review by jgilliam17 — APPROVED** [review]
Thanks @santoniriccardo 
I ran the [E2Es](https://mattermost-cypress-report.s3.amazonaws.com/208299-e1a88f5-onprem-ent-server-pr-20259-mattermost/mm-ee-teste1a88f5/mochawesome.html) and will test the changes on the related webapp PR after merge.

**Pull request merged**